### PR TITLE
separates provider and common setup into files

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: integration/setup/common.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+	corev1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/e2e-harness/pkg/release"
+	"github.com/giantswarm/e2etemplates/pkg/chartvalues"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/azure-operator/integration/env"
+)
+
+func Common(config Config) error {
```
Comment (TheoBrigitte) [inline integration/setup/common.go]: A comment on what does this function would be nice.

File: integration/setup/provider.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/e2e-harness/pkg/release"
+	"github.com/giantswarm/e2etemplates/pkg/chartvalues"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/azure-operator/integration/env"
+)
+
+func Provider(config Config) error {
```
Comment (TheoBrigitte) [inline integration/setup/provider.go]: A comment on what does this function would be nice.

File: integration/setup/common.go (hunk 1 above)
Comment (xh3b4sd) [reply integration/setup/common.go]: Then please provide a suggestion. This is so obvious to me personally that I do not know what to write here as everything I could come up with would be rather useless. Please also suggest for the one below.

[the remark this answers is not in the document]
Comment (xh3b4sd) [reply integration/setup/common.go]: Done.

**Comment by xh3b4sd** [comment]
Going with this. @TheoBrigitte feel free to suggest godoc I will add it then later.

File: integration/setup/common.go (hunk 1 above)
Comment (TheoBrigitte) [reply integration/setup/common.go]: common installs components required to run the operator.

provider installs the operator and tenant cluster CR.

**Pull request merged**